### PR TITLE
feature/1444 - Add ability to unamend an unmodified amendment

### PR DESCRIPTION
**Pull request by sasha-dresden**
Issue [FECFILE-1444](https://fecgov.atlassian.net/browse/FECFILE-1444)
API PR [PR1032](https://github.com/fecgov/fecfile-web-api/pull/1032)

### CODE REVIEW

File: front-end/src/app/reports/report-list/report-list.component.ts [hunk 1]
```diff
@@ -107,6 +108,28 @@ export class ReportListComponent extends TableListBaseComponent<Report> implemen
       });
   }
 
+  public confirmUnamend(report: Report): void {
```
Comment (toddlees) [inline front-end/src/app/reports/report-list/report-list.component.ts]: I'm not seeing a confirm dialog in the ticket.  I think we need to leave this out.

File: front-end/src/app/reports/report-list/report-list.component.ts [hunk 2]
```diff
@@ -107,6 +108,28 @@ export class ReportListComponent extends TableListBaseComponent<Report> implemen
...
+      accept: async () => this.unamendReport(report),
+    });
+  }
+
+  async unamendReport(report: Report) {
+    this.itemService
+      .startUnamendment(report)
+      .pipe(take(1), takeUntil(this.destroy$))
+      .subscribe(() => {
+        this.loadTableItems({});
```
Comment (toddlees) [inline front-end/src/app/reports/report-list/report-list.component.ts]: The ticket references a toast message.  Probably need to add:
```
    this.messageService.add({
      severity: 'success',
      summary: 'Successful',
      detail: 'Report Unamended',
      life: 3000,
    });
```

File: front-end/src/app/shared/services/report.service.ts [hunk 3]
```diff
@@ -104,6 +104,10 @@ export class ReportService implements TableListService<Report> {
     return this.apiService.post(`${this.apiEndpoint}/${report.id}/amend/`, {});
   }
 
+  public startUnamendment(report: Report): Observable<string> {
```
Comment (toddlees) [inline front-end/src/app/shared/services/report.service.ts]: I see we're following the pattern the amendment implementation set.  What do you think of changing these two to promises? 

```
public amend(report:Report): Promise<string> {
  return getFirstValueFrom(this.apiService.post(...))
 }
```

**Review by toddlees — CHANGES_REQUESTED** [review]

File: front-end/src/app/reports/report-list/report-list.component.ts (hunk 1 above)
Comment (sasha-dresden) [reply front-end/src/app/reports/report-list/report-list.component.ts]: Took out confirm.

File: front-end/src/app/reports/report-list/report-list.component.ts (hunk 2 above)
Comment (sasha-dresden) [reply front-end/src/app/reports/report-list/report-list.component.ts]: Added in toast.

File: front-end/src/app/shared/services/report.service.ts (hunk 3 above)
Comment (sasha-dresden) [reply front-end/src/app/shared/services/report.service.ts]: Oh I just saw this one. Yea promises would make a lot more sense here.

**Review by toddlees — APPROVED** [review]

**Pull request merged**